smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelAssembler.java

```diff
@@ -246,13 +248,21 @@ public ModelAssembler addImport(Path importPath) {
 
         if (Files.isDirectory(importPath)) {
             try {
-                Files.walk(importPath)
+                Files.walk(importPath, FileVisitOption.FOLLOW_LINKS)
```

Comment:
After pulling this down and testing, I think this is actually the only line that needs to change. I was able to get the added test to pass by removing every other change but this one. Any reason for the other changes?

Reply:
The change comes into play when we pass a singular symlink targets instead of a directory. I will add appropriate test for this case as well by EOD

Reply:
That would be great. The setting of NOFOLLOW_LINKS and the introduced or statement to check if a file is a symbolic link shouldn't be necessary since `Files.isRegularFile` should automatically follow symlinks: The setting of NOFOLLOW_LINKS and the introduced or statement to check if a file is a symbolic link shouldn't be necessary since `Files.isRegularFile` should automatically follow symlinks: https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#isRegularFile-java.nio.file.Path-java.nio.file.LinkOption...-

Reply:
Yes you are right with your point of not having to isolate logic for detecting symlinks,  ```Files.isRegularFile``` should be sufficient to detect both symlinks and regular files. I did that in rev#1 since I was isolating the logic for importing regular files and symlinks. I have removed the extra check in rev#2.

So there are 2 use cases,

- **Symlinks under a directory / subdirectory**
In this case ```FileVisitOption.FOLLOW_LINKS``` change in files walk resolve the symlinks to actual target path and recursively trigger the logic for importing a regular file. Regular file import works fine since the import path = target path and does not require any additional change in the regular file import logic. For this I have added ```importsSymlinksDirectoryWithAllShapes``` test.

- **Single symlink file**
In this case the original import logic for regular file is not sufficient. This is because in this we try to import using the symlink (```importPath```) and not the actual target path. Therefore it is either required to have different import logic for symlink file (rev#1) or change the current regular file import logic to accommodate symlinks. In rev#2 I have gone with the latter and used ```importPath.toRealPath()``` instead of plain ```importPath```. This is because ```toRealPath()``` [works for both regular file and symlink](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Path.html#toRealPath(java.nio.file.LinkOption...)), for symlink this method yields the target path. To test this I have added ```importsSymlinkFileWithAllShapes``` test to confirm. If ```importPath.toRealPath()``` is removed, this test fails as importPath=symlink and nothing is imported.

In Rev#1 I covered both of the above cases by isolating the import logic for symlinks and regular files since I did not want to touch the regular file import logic. In Rev#2, it's the same but now I augmented the regular file import logic to accommodate symlinks by using ```importPath.toRealPath()``` instead of plain ```importPath```. Latter should be okay since we have good tests covering regular file imports as well.